main/src/resources/doc/index.js

```diff
@@ -0,0 +1,31 @@
+const storeKey = "flix-html-docs:use-dark-theme";
+
+const body = document.querySelector("body");
+body.classList.remove("no-script");
+
+function setTheme(dark) {
+    if (dark) {
+        body.classList.remove("light");
+        body.classList.add("dark");
+    } else {
+        body.classList.remove("dark");
+        body.classList.add("light");
+    }
+}
+
+let dark;
+const stored = localStorage.getItem(storeKey);
+if (stored === null) {
+    dark = window.matchMedia("(prefers-color-scheme: dark)").matches;
```

Comment:
// User has not stored any preference. Check if the browser prefers a dark theme.